pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -376,8 +376,7 @@ func (p *Planner) Process(controlPlane *rkev1.RKEControlPlane) error {
 		return ErrWaiting(firstIgnoreError.Error())
 	}
 
-	if controlPlane.Spec.RotateEncryptionKeys != nil &&
-		controlPlane.Status.RotateEncryptionKeysGeneration != controlPlane.Spec.RotateEncryptionKeys.Generation {
+	if rke2.Updated.IsUnknown(controlPlane) && rke2.Updated.GetReason(controlPlane) == rke2.RotatingEncryptionKeysReason {
```

Comment:
I'm reading this as- before, if we aren't on the correct rotation key generation then requeue the rotation. After your change, if the cp node has not finished updating because encryption key rotation is in progress, requeue the rotation. This doesn't sound right. Why do we need to requeue if the current rotation just isn't finished? (or help me understand the logic here).

Reply:
I think I'm following. So this code

`if rke2.Updated.IsUnknown(controlPlane) && rke2.Updated.GetReason(controlPlane) == rke2.RotatingEncryptionKeysReason`

Basically, checks that if the cp node went down and RotatingEncryptionKeysReason is set (meaning the cp node crashed during encryption key rotation), then requeue the rotation?